engine/src/board.rs

```diff
@@ -172,10 +186,9 @@ impl Board {
             .filter(|pos| {
                 self.board
                     .get(pos)
-                    .unwrap() // unwrap here is safe because we are got pos from board.keys
+                    .expect("Board has the key, what is wrong here?!")
```

Comment:
you might consider rewriting this function to iterate over keys/values at the same time, like:

```
    pub fn positions_for_color(&self, color: &Color) -> Vec<Position> {
        self.board.iter()
            .filter_map(|(pos, pieces)| {
                if let Some(piece) = pieces.last() {
                    if piece.is_color(color) {
                        return Some(*pos);
                    }
                }
                None
            })
            .collect()
    }
```